drivers/gles3/storage/mesh_storage.cpp

```diff
@@ -259,11 +259,19 @@ void MeshStorage::mesh_add_surface(RID p_mesh, const RS::SurfaceData &p_surface)
 			mesh->bone_aabbs.resize(p_surface.bone_aabbs.size());
 		}
 		for (int i = 0; i < p_surface.bone_aabbs.size(); i++) {
-			const AABB &bone = p_surface.bone_aabbs[i];
-			if (!bone.has_no_volume()) {
-				mesh->bone_aabbs.write[i].merge_with(bone);
+			AABB &mesh_bone = mesh->bone_aabbs.write[i];
+			const AABB &surface_bone = p_surface.bone_aabbs[i];
+			if (!surface_bone.has_no_volume()) {
+				if (mesh_bone.has_no_volume()) {
```

Comment:
I would compare with `AABB()` instead since otherwise it may not have a volume and have offset because its a single vertex and in that case the vertex would be ignored.

Reply:
Comparing against `AABB()` could result in ignoring `(0, 0, 0)` vertex in the same manner though.

Reply:
I updated the code to resolve reduz's original idea, but maybe we need to designate `AABB((0, 0, 0), (-1, -1, -1))` as uninitialized?

Reply:
> Comparing against `AABB()` could result in ignoring `(0, 0, 0)` vertex in the same manner though.

When creating the bone array, AABBs always have a slightly positive size: https://github.com/godotengine/godot/blob/051f24b067642a241d0a4acbb1b7644e11700bfc/servers/rendering_server.cpp#L687 
So they won't be equal to ``AABB()``